### PR TITLE
Add basic Rocket list classes.

### PR DESCRIPTION
**Pull request by alesrazym**

### CODE REVIEW

**Comment by alesrazym** [comment]
@havlisimo I've added basic classes, as discussed in person.

Will continue with `Koin`, probably in a separate branch so that we can close PRs.

File: app/src/main/java/cz/quanti/razym/rocketapp/domain/RocketsRepo.kt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package cz.quanti.razym.rocketapp.domain
+
+interface RocketsRepo {
```
Comment (havlisimo) [inline app/src/main/java/cz/quanti/razym/rocketapp/domain/RocketsRepo.kt]: mozna jsem to blbe napsal na ten papir, ale pouzivame cele slovo Repository pro interface i implementaci

File: app/src/main/res/layout/activity_rocket.xml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
+<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (havlisimo) [inline app/src/main/res/layout/activity_rocket.xml]: lepsi je pouzit FragmentContainerView

File: app/src/main/java/cz/quanti/razym/rocketapp/system/RocketListFragment.kt [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+            recycler.layoutManager = LinearLayoutManager(requireContext())
+            recycler.adapter = RocketListAdapter(rockets)
+        }
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        return inflater.inflate(R.layout.fragment_rocket_list, container, false)
```
Comment (havlisimo) [inline app/src/main/java/cz/quanti/razym/rocketapp/system/RocketListFragment.kt]: pouzij view binding misto rucniho inflatovani a pouzivani findViewById
https://developer.android.com/topic/libraries/view-binding

File: app/src/main/java/cz/quanti/razym/rocketapp/domain/RocketModel.kt [hunk 4]
```diff
@@ -0,0 +1,3 @@
+package cz.quanti.razym.rocketapp.domain
```
Comment (havlisimo) [inline app/src/main/java/cz/quanti/razym/rocketapp/domain/RocketModel.kt]: model by mel byt v package model (to jsem ti asi vcera rekl spatne) a nemel by obsahovat slovo Model, staci Rocket

File: app/src/main/java/cz/quanti/razym/rocketapp/presentation/RocketListAdapter.kt [hunk 5]
```diff
@@ -0,0 +1,39 @@
+package cz.quanti.razym.rocketapp.presentation
```
Comment (havlisimo) [inline app/src/main/java/cz/quanti/razym/rocketapp/presentation/RocketListAdapter.kt]: patry do system vrstvy

File: app/src/main/java/cz/quanti/razym/rocketapp/system/RocketListFragment.kt [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    private val viewModel: RocketListViewModel by activityViewModels {
+        RocketListViewModel.RocketListViewModelFactory(requireContext())
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        viewModel.rocketLiveData.observe(this) { rockets ->
+            val recycler = requireView().findViewById<RecyclerView>(R.id.rocket_list)
+            recycler.layoutManager = LinearLayoutManager(requireContext())
```
Comment (havlisimo) [inline app/src/main/java/cz/quanti/razym/rocketapp/system/RocketListFragment.kt]: lze nastavit v xml 
        app:layoutManager="androidx.recyclerview.widget.LinearLayoutManager"

File: app/src/main/res/layout/rocket_list_item.xml [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+        android:id="@+id/rocket_item_subtitle"
+        android:layout_width="0dp"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="16dp"
+        app:layout_constraintBottom_toBottomOf="@+id/rocket_item_image"
+        app:layout_constraintEnd_toStartOf="@id/barrier2"
+        app:layout_constraintStart_toEndOf="@id/barrier1"
+        tools:text="First Flight 2006-03-24" />
+
+    <androidx.constraintlayout.widget.Barrier
```
Comment (havlisimo) [inline app/src/main/res/layout/rocket_list_item.xml]: obe bariery mi prijdou nadbytecne

ikonu rakety zarovnas podle parenta na start, top, bottom
ikonu chevronu podle parenta na end, top, bottom
a textviews mezi ne

**Review by havlisimo — APPROVED** [review]

**Pull request merged**